java/src/jmri/jmrix/AbstractSerialPortController.java

```diff
@@ -413,19 +412,28 @@ final protected FlowControl getFlowControl(com.fazecast.jSerialComm.SerialPort s
 
     /**
      * Add a data listener to the specified port
-     * @param serialPort Port to be updated
      * @param serialPortDataListener the listener to add
      */
-    final protected void setDataListener(com.fazecast.jSerialComm.SerialPort serialPort,com.fazecast.jSerialComm.SerialPortDataListener serialPortDataListener){
-        currentSerialPort.addDataListener(serialPortDataListener);
+    final protected void setDataListener(SerialPortDataListener serialPortDataListener){
+        currentSerialPort.serialPort.addDataListener(new com.fazecast.jSerialComm.SerialPortDataListener() {
+            @Override
+            public int getListeningEvents() {
+                return serialPortDataListener.getListeningEvents();
+            }
+
+            @Override
+            public void serialEvent(com.fazecast.jSerialComm.SerialPortEvent event) {
+                serialPortDataListener.serialEvent(new SerialPortEvent(event));
+            }
+        });
     }
```

Comment:
While I was working on this PR, I discovered a bug. This method had a parameter `serialPort`, but that parameter wasn't used. I wasn't sure how to deal with it and I didn't want to create a bug, so I removed the parameter.

But to be consistent with other methods, that parameter might be restored and the method changed so that it adds the listener to the `serialPort` parameter, instead to the `currentSerialPort`?

Reply:
After a second look, I noticed that the original parameter sent to the method was `currentSerialPort`. So I have restored the parameter to make the method consistent with other methods. And I fixed the bug so the parameter is used now.